action.yml

```
@@ -0,0 +1,25 @@
+name: 'Mariachi'
+description: 'Sends Pull Request review reminders to Microsoft Teams channels'
+inputs:
+  github_token:
+    description: 'This is the default token set and created by GitHub. you can also use a personal access token with repo scope enabled, but this one is already available'
+    required: false
+    default: ${{ github.token }}
+  teams_url: 
+    description: "your Teams channel's webhook URL. here we assume it's in your repo's secrets store"
+    required: true
+  exclude_heads:
+    description: 'Head Git branches to exclude'
+    required: false
+  exclude_labels:
+    description: 'Lables to exclude'
+    required: false
+  min_reviews:
+    description: 'Minimum number of reviews needed'
+    required: false
+runs:
+  using: 'docker'
+  image: 'Dockerfile'
```

This looks like it should point to the precompiled Mariachi docker image:
```suggestion
  image: 'docker://schlagelk/mariachi:latest'
```

---

Ah gotcha. I wondered if there would be a chance if I'm using the `@main` version of the action, and a new commit lands, but the Dockerhub image hasn't been pushed yet. Direct support for the docker file seemed to be well supported via https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action and https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsimage. Other actions I've seen in the marketplace generally use the local Dockerfile as well instead of a DockerHub URL.

---

That's a good thought, let's try it this way and see how it goes. If and when I get to pushing up a new docker image on commits to main I can revisit.